File: app/src/main/java/com/example/challenge_consult_cep/feature/search_cep/data/mapper/ViaCepDTOMapper.java
package com.example.challenge_consult_cep.feature.search_cep.data.mapper;

import com.example.challenge_consult_cep.feature.search_cep.data.remote.dto.ViaCepDTO;
import com.example.challenge_consult_cep.feature.search_cep.domain.model.Cep;

public class ViaCepDTOMapper {
    public static Cep toCep(ViaCepDTO viaCepDTO) {
        return new Cep(viaCepDTO.getCep().replace("-", ""), viaCepDTO.getUf(), viaCepDTO.getLocalidade(), viaCepDTO.getBairro(), "Rua " + viaCepDTO.getLogradouro());
    }
}

File: app/src/main/java/com/example/challenge_consult_cep/feature/search_cep/data/mapper/ViaCepDTOMapper.kt
package com.example.challenge_consult_cep.feature.search_cep.data.mapper

import com.example.challenge_consult_cep.feature.search_cep.data.remote.dto.ViaCepDTO
import com.example.challenge_consult_cep.feature.search_cep.domain.model.Cep

fun ViaCepDTO.toCep() = Cep(
    code = cep,
    state = uf,
    city = localidade,
    neighborhood = bairro,
    address = "Rua $logradouro",
)